### PR TITLE
appsec: integrate the new purego-based go-libddwaf

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

This go-libddwaf version now relies on the dynamic liddwaf library instead of the static one, via `dlopen()` and `dlsym()` internally.
This allows us to no longer rely on CGO at compilation time, with all the requirements it needed such as a proper C toolchain. To do so, we now rely on https://github.com/ebitengine/purego which mainly provides a high-level API on top of `runtime/cgo` to call the C world.

The overall API of the package didn't change, but we split the former `Health()` function in two to better integrate it into dd-trace-go and handle the different failure points correctly:
1. Log an error and quit appsec if we are in a host environment not supported by libddwaf, as early as possible, avoiding for instance to start remote configuration.
2. Log an error and quit appsec in case of unexpected errors when loading libddwaf (eg. if the filesystem is read-only and we can't dump the libddwaf.so file)
3. Log and error but keep going in case of unexpected non-critical errors.

### Motivation

No more CGO prerequisites, mainly:
1. C toolchain, with extra extra complexity when cross-compiling (even on simple cases like from a debian (glibc-based) to alpine (musl-based) 
2. CGO_ENABLED=1

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Comment by eliottness** [comment]
Should we wait until the go-libddwaf final PR is merged to merge this one ?

File: internal/appsec/remoteconfig.go
Comment (Hellzy) [inline internal/appsec/remoteconfig.go:337]:
```suggestion
		return fmt.Errorf("appsec: no valid remote configuration client")
```

Sorry I know this isn't part of your changes

File: internal/appsec/appsec.go
Comment (Hellzy) [inline internal/appsec/appsec.go:85]:
```suggestion
		log.Debug("appsec: %s is not set. ASM won't start until activated through remote configuration", enabledEnvVar)
```
Otherwise it feels like we're saying the env var won't start. Feel free to replace `ASM` by anything else you deem more fitting

**Review by Hellzy — COMMENTED** [review]
LGTM overall, a a few comments about debug logs

File: internal/appsec/appsec.go
Comment (eliottness) [reply internal/appsec/appsec.go:52]: It's is not a big deal until we remove the appsec build tag but still

Comment (Hellzy) [reply internal/appsec/appsec.go:52]: I don't think there's any value in waiting for a remote activation update, when we can error out at the very start of appsec. Basically what this does is error out on appsec.Start() if the WAF is not useable, instead of erroring out in the RC update that activates appsec in case of remote activation. No sense in "allowing" remote activation if we know it's going to fail.

Comment (eliottness) [reply internal/appsec/appsec.go:52]: Yes we want to do as most check as possible early on. But do we want to have the WAF loaded (and it's memory footprint) even if appsec will probably never be enabled ?

Comment (Julio-Guerra) [reply internal/appsec/appsec.go]: Should be fixed by https://github.com/DataDog/dd-trace-go/pull/2090/commits/4dbdfb0b11df37e4d8174259dd748d09592b6005

File: internal/appsec/appsec.go
Comment (Julio-Guerra) [reply internal/appsec/appsec.go:73]: I think appsec if better to use it the lib and the logs as this is our the prefix we use and ASM really is too ambiguous for devs.

File: internal/appsec/appsec.go
Comment (Julio-Guerra) [reply internal/appsec/appsec.go]: Fixed in 7b033143

**Review by Hellzy — APPROVED** [review]

**Review by katiehockman — APPROVED** [review]
Didn't review the code, but the overall idea looks good. This is so awesome!
Approving just based on the `go.mod` and `go.sum` files. LMK if you want another set of eyes on the `internal/appsec` changes you made here, too, and I can check it out.

**Pull request merged**
